hw/xbox/nv2a.c

```diff
@@ -1866,10 +1892,49 @@ static TextureBinding* generate_texture(const TextureShape s,
                 unswizzle_rect(texture_data, width, height,
                                unswizzled, pitch, f.bytes_per_pixel);
 
+                uint8_t *data = unswizzled;
+  
+                if (s.color_format == NV097_SET_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8) {
+                  data = g_malloc(height * pitch * 4);
+                  unsigned int y;
+                  for(y = 0; y < height; y++) {
+                    unsigned int x;
+                    for(x = 0; x < pitch; x++) {
+                      uint8_t index = unswizzled[y * pitch + x];
+                      uint8_t* pixel = &data[(y * pitch + x) * 4];
+                      const uint8_t* color = &palette_data[index * 4];
+                      pixel[0] = color[0];
+                      pixel[1] = color[1];
+                      pixel[2] = color[2];
+                      pixel[3] = color[3];
+                    }
+                  }
+                }
+
+                if (s.color_format == NV097_SET_TEXTURE_FORMAT_COLOR_LC_IMAGE_CR8YB8CB8YA8) {
+                  data = g_malloc(height * pitch * 4);
+                  unsigned int y;
+                  for(y = 0; y < height; y++) {
+                    unsigned int x;
+                    for(x = 0; x < pitch; x++) {
+                      uint8_t* pixel = &data[(y * pitch + x) * 4];
+                      //FIXME: Steal format conversion from video stuff above
```

Comment:
I'll add the new handler to this PR and move the video YUV conversion to a more generic color function